src/main/java/org/beanpod/switchboard/dao/StreamDaoImpl.java

```diff
@@ -58,4 +71,25 @@ public void updateStream(StreamDTO streamDto) {
     StreamEntity streamEntity = mapper.toEntity(streamDto);
     streamRepository.save(streamEntity);
   }
+
+  private boolean onSamePrivateNetwork(
+      InputChannelDTO inputChannelDto, OutputChannelDTO outputChannelDto) {
+    return inputChannelDto
+        .getDecoder()
+        .getDevice()
+        .getPublicIpAddress()
+        .equals(outputChannelDto.getEncoder().getDevice().getPublicIpAddress());
+  }
+
+  private boolean deviceOnLocalNetwork(DeviceDTO deviceDto) {
+    return deviceDto.getPublicIpAddress().equals(deviceDto.getPrivateIpAddress())
+        || deviceDto.getPublicIpAddress().equals(LOOPBACK_IP_V4)
+        || deviceDto.getPublicIpAddress().equals(LOOPBACK_IP_V6);
+  }
+
+  private boolean onLocalNetwork(
+      InputChannelDTO inputChannelDTO, OutputChannelDTO outputChannelDTO) {
+    return deviceOnLocalNetwork(inputChannelDTO.getDecoder().getDevice())
+        && deviceOnLocalNetwork(outputChannelDTO.getEncoder().getDevice());
+  }
```

Comment:
I think this would be a good place to put a comment or two explaining how we determine if devices are on local networks, and why that affects our decision to use SRT mode or not.

Reply:
Would you rather have `publicIpAddress` name `requestIpAddress` because in reality it's the request that we grab from the HTTP request.

Reply:
I added comments on the two methods you requested. I also moved them to `NetworkingUtil` and for some reason I feel a lot more comfortable adding comments to util classes rather than controller/service/dao classes.